src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportBasicService.java

```diff
@@ -211,6 +218,31 @@ private void setupTestRepository(ProgrammingExercise newExercise) {
         newExercise.setTestRepositoryUri(versionControlService.orElseThrow().getCloneRepositoryUri(newExercise.getProjectKey(), testRepoName).toString());
     }
 
+    private void setupBuildConfig(ProgrammingExercise newExercise, ProgrammingExercise originalExercise) {
+        if (newExercise.getBuildConfig() != null) {
+            var buildConfig = newExercise.getBuildConfig();
+            buildConfig.setId(null);
+            buildConfig.setProgrammingExercise(null);
+            newExercise.setBuildConfig(buildConfig);
+        }
+        else if (originalExercise.getBuildConfig() != null) {
+            var buildConfig = new ProgrammingExerciseBuildConfig();
+            buildConfig.setBranch(originalExercise.getBuildConfig().getBranch());
+            buildConfig.setBuildPlanConfiguration(originalExercise.getBuildConfig().getBuildPlanConfiguration());
+            buildConfig.setCheckoutPath(originalExercise.getBuildConfig().getCheckoutPath());
+            buildConfig.setCheckoutSolutionRepository(originalExercise.getBuildConfig().getCheckoutSolutionRepository());
+            buildConfig.setDockerFlags(originalExercise.getBuildConfig().getDockerFlags());
+            buildConfig.setSequentialTestRuns(originalExercise.getBuildConfig().hasSequentialTestRuns());
+            buildConfig.setBuildScript(originalExercise.getBuildConfig().getBuildScript());
+            buildConfig.setTestwiseCoverageEnabled(originalExercise.getBuildConfig().isTestwiseCoverageEnabled());
+            buildConfig.setTimeoutSeconds(originalExercise.getBuildConfig().getTimeoutSeconds());
+            newExercise.setBuildConfig(buildConfig);
```

Comment:
Can you move this into a copy method in BuildConfig? Otherwise this place might easily be overlooked when adding a new attribute